modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateMSA_conf.pm

```diff
@@ -0,0 +1,442 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::PipeConfig::UpdateMSA_conf
+
+=head1 SYNOPSYS
+
+    init_pipeline.pl Bio::EnsEMBL::Compara::PipeConfig::UpdateMSA_conf -host mysql-ens-compara-prod-X -port XXXX \
+        -division $COMPARA_DIV -method_type <lowercase_method_type> -species_set_name <species_set_name>
+
+=head1 DESCRIPTION
+
+The PipeConfig file for the pipeline that copies the data from the previous
+release MSA and updates it. This update involves:
+- Mercator-Pecan: remove the updated species and recompute GERP
+- EPO: copy the ancestral core db and update the dna names, remove the updated
+  species from the EPO MSA and update the ancestral dnafrag ids, and recompute
+  its EPO Extended MSA (GERP included), incorporating the updated species
+
+=cut
+
+package Bio::EnsEMBL::Compara::PipeConfig::UpdateMSA_conf;
+
+use strict;
+use warnings;
+
+use JSON;
+
+use Bio::EnsEMBL::Hive::PipeConfig::HiveGeneric_conf;  # for WHEN and INPUT_PLUS
+
+use Bio::EnsEMBL::Compara::PipeConfig::Parts::EPOAncestral;
+use Bio::EnsEMBL::Compara::PipeConfig::Parts::EpoLowCoverage;
+
+use base ('Bio::EnsEMBL::Compara::PipeConfig::ComparaGeneric_conf');
+
+
+sub default_options {
+    my ($self) = @_;
+    return {
+        %{$self->SUPER::default_options},  # inherit the generic ones
+
+        'pipeline_name' => $self->o('species_set_name') . '_' . $self->o('method_type') . '_update_' . $self->o('rel_with_suffix'),
+        'work_dir'      => $self->o('pipeline_dir'),
+        'output_dir'    => $self->o('work_dir') . '/feature_dumps/',
+        'bed_dir'       => $self->o('work_dir') . '/bed_dir/',
+
+        'master_db'         => 'compara_master',
+        'prev_db'           => 'compara_prev',
+        'reuse_db'          => $self->o('species_set_name') . '_' . $self->o('method_type') . '_prev',
+        'prev_ancestral_db' => $self->o('species_set_name') . '_ancestral',
+        
+        # EPOAncestral parameters
+        'ancestral_sequences_name'          => 'ancestral_sequences',
+        'ancestral_sequences_display_name'  => 'Ancestral sequences',
+        # Core ancestral database, created on the same server as the pipeline database
+        'ancestral_db'             => {
+            -driver  => $self->o('pipeline_db', '-driver'),
+            -host    => $self->o('pipeline_db', '-host'),
+            -port    => $self->o('pipeline_db', '-port'),
+            -user    => $self->o('pipeline_db', '-user'),
+            -pass    => $self->o('pipeline_db', '-pass'),
+            -species => $self->o('ancestral_sequences_name'),
+            -dbname  => $self->o('dbowner') . '_' . $self->o('species_set_name') . '_ancestral_core_' . $self->o('rel_with_suffix'),
+        },
```

Comment:
This is a copy-paste from `EPOwith2x`: https://github.com/Ensembl/ensembl-compara/blob/release/103/modules/Bio/EnsEMBL/Compara/PipeConfig/EPOwith2x_conf.pm#L100-L108
I guess it is done this way so you don't need to configure the registry *before* the pipeline has finished, as we usually do with all the other pipelines.

Reply:
(yes, that was the reason)

Reply:
Adding more views to the discussion ...

There is a difference though: currently we only edit the required _input_ databases in the registry file before starting a pipeline, and the location of the pipeline (i.e. its output) is set on the `init_pipeline.pl` command-line. This `ancestral_db` is the rare case of a pipeline having a second _output_ database which, as it is/was, draws its parameters from the same command-line options.
There is the case of `compara_curr`, which several pipelines used for (incremental) output, and is assumed to be there in the registry file ... **But** it's a _shared_ output database, as opposite to the ancestral db which entirely comes from this pipeline.

I also think that, when doing a test run of the pipeline, there is a higher risk of erasing an _existing_ ancestral database if it uses the registry file instead of creating its own, new database.

:shrug:

Reply:
Two more things to consider

1. I seem to remember that loading the registry file will _fail_ if a database doesn't exist, so it can't be used to tell where to create a database
2. In the pipeline the location of the ancestral db is actually primarily stored in `genome_db.locator`. Once the genome_db has been created, the source of the information (pipeline param or registry file doesn't matter any more).  I see that as a risk in the sense that when doing a test run, I may forget about that, edit the registry file, and start my tests without realising the tests will modify the true ancestral database

Reply:
Well I think that's the nail in the coffin for setting it in the registry - let's leave it as is.